packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/GroupHiding.ts

```diff
@@ -8,30 +8,25 @@ import { GroupingHandlerResult, GroupingType } from "../Grouping";
 
 /** @internal */
 export function applyGroupHidingParams(props: GroupingHandlerResult): GroupingHandlerResult {
-  if (props.groupedNodes.length === 0) {
+  if (props.grouped.length === 0) {
     return props;
   }
-  const finalGroupings: GroupingHandlerResult = { allNodes: [], groupedNodes: [], groupingType: props.groupingType };
-  let lastGroupingNodeIndex = 0;
-  for (const node of props.allNodes) {
-    if (props.groupedNodes.length <= lastGroupingNodeIndex || node !== props.groupedNodes[lastGroupingNodeIndex]) {
-      finalGroupings.allNodes.push(node);
-      continue;
-    }
-    ++lastGroupingNodeIndex;
-    if (Array.isArray(node.children) && (props.allNodes.length === 1 || node.children.length === 1)) {
+  const finalGroupings: GroupingHandlerResult = { grouped: [], ungrouped: [], groupingType: props.groupingType };
```

Comment:
```suggestion
  const finalGroupings = { ...props, grouped: [] };
```

and then we don't need the `finalGroupings.ungrouped.push(...props.ungrouped);` line at the bottom